src/core/unit.c

```diff
@@ -3112,6 +3112,23 @@ int unit_add_dependency(
                         return r;
                 if (r)
                         noop = false;
+
+                /* If the inverse dependency includes the REFERENCES atom then
+                 * add a REFERENCES/REFEENCED_BY dependencies to keep the
+                 * REFERENCED dependency pinned. */
+                if (FLAGS_SET(unit_dependency_to_atom(inverse_table[d]), UNIT_ATOM_REFERENCES)) {
```

Comment:
hmm, i think instead we should probably define yet another atom UNIT_ATOM_BACK_REFERENCE_IMPLIED or so, which controls this, and doesn't require us to manually follow te inversion table?

Reply:
Yep, makes sense. Next revision uses a new atom type.